app/models/security_activity.rb

```diff
@@ -62,11 +64,21 @@ def self.record_event(event, options = {})
       attributes.merge!(user_agent_id: UserAgent.find_or_create_by!(name: options[:user_agent_name]).id)
     end
 
+    if attributes[:ip_address] && !attributes[:ip_address_country]
+      attributes[:ip_address_country] = ip_to_country(attributes[:ip_address])
+    end
+
     event = SecurityActivity.create!(attributes)
     event.log
     event
   end
 
+  def self.ip_to_country(ip_address)
+    Rails.cache.fetch("security_activities/ip_address/#{ip_address}", expires_in: 1.day) do
+      Geocoder.search(ip_address)&.first&.country
```

Comment:
Is there a chance that `.search` failing could prevent the security event being recorded? Do you think this should be wrapped in a rescue block.

Reply:
Geocoder swallows all the exceptions unless we explicitly turn them on: https://github.com/alexreisner/geocoder/#error-handling